databricks_cli/stack/api.py

```diff
@@ -0,0 +1,349 @@
+
+        return parsed_conf
+
+    def _handle_json_datetime(self, obj):
+        """
+        Helper function to convert datetime into timestamp. Raise a regular JSON TypeError if
+        a datetime object is not given.
+
+        :param obj: Object that may be a datetime object.
+        :return: Timestamp integer if object is a datetime object.
+        """
+        if type(obj) is datetime:
+            # Get timestamp of datetime object- works with python2 and 3
+            return int(time.mktime(datetime.now().timetuple()))
+        raise TypeError("Object of type '%s' is not JSON serializable" % type(obj))
+
+    def _generate_stack_status_path(self, stack_path):
+        """
+        Given a path to the stack configuration template JSON file, generates a path to where the
+        deployment status JSON will be stored after successful deployment of the stack.
+
+        :param stack_path: Path to the stack config template JSON file
+        :return: The path to the stack status file.
+
+        >>> self._generate_stack_status_path('./stack.json')
+        './stack.deployed.json'
+        """
+        stack_path_split = stack_path.split('.')
+        stack_path_split.insert(-1, STACK_STATUS_INSERT)
+        return '.'.join(stack_path_split)
+
+    def _load_stack_status(self, status_path):
+        """
+        Loads the deployment status metadata for a stack given a path to the stack configuration
+        template JSON file. Looks for the default local stack status path generated from
+        _generate_stack_status_path.
+
+        When loaded, the stack resource configurations from the past deployment will be loaded into
+        self.deployed_resource_config, using the RESOURCE_ID field of each resource as a key in the
+        dictionary.
+        The output from the databricks server of the deployment of each resource will also be
+        loaded in self.deployed_resources in the same way.
+
+        :param status_path: path to JSON stack configuration template.
+        :return: The dict of parsed JSON of the stack deployment status.
+        If path doesn't exist, will return an empty dict.
+        """
+        parsed_conf = {}
+        try:
+            if os.path.exists(status_path):
+                with open(status_path, 'r') as f:
+                    parsed_conf = json.load(f)
+                click.echo("Using deployment status file at %s" % status_path)
+        except ValueError:
+            # Handles a bad JSON read. Will just pass and parsed_conf will be empty dict.
+            pass
+
+        if STACK_RESOURCES in parsed_conf:
+            self.deployed_resource_config = parsed_conf[STACK_RESOURCES]
+        if STACK_DEPLOYED in parsed_conf:
+            self.deployed_resources = {resource[RESOURCE_ID]: resource for resource in
+                                       parsed_conf[STACK_DEPLOYED]}
+
+        return parsed_conf
+
+    def _get_deployed_resource(self, resource_id, resource_service):
```

Comment:
I want to catch whether a deployed resource does not match the resource and return None so that the deployment code doesn't need to deal with a deployed resource that has a different service than requested.